src/App.tsx

```diff
@@ -1,19 +1,149 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.css';
-
+import todosFromServer from './api/todos';
 import users from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
+
+const preparedData: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: users.find((user) => user.id === todo.userId) || null,
+}));
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todos: Todo[];
+  title: string;
+  userId: number;
+  userSelected: boolean;
+  titleSelected: boolean;
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todos: preparedData,
+    title: '',
+    userId: 0,
+    userSelected: true,
+    titleSelected: true,
+  };
+
+  addTodo = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+    const { userId, title } = this.state;
+
+    this.setState(prevState => {
+      const createdTodo = {
+        user: users.find(user => user.id === userId) || null,
+        userId,
+        id: Math.max(...prevState.todos.map(item => item.id)) + 1,
+        title,
+        completed: false,
+      };
+
+      // eslint-disable-next-line no-console
+      console.log(createdTodo);
+
+      return {
+        todos: [...prevState.todos, createdTodo],
+        userId: 0,
+        title: '',
+      };
```

Comment:
it should be done only if todo was added. Because now both fields get clear, even if empty only one field
https://www.loom.com/share/59582ad574704c26b958746f969e0c2a